client/mussel/v12.03.d/instance.sh

```diff
@@ -132,6 +132,37 @@ task_show_password() {
   call_api -X GET $(base_uri)/${namespace}s/${uuid}/password.$(suffix)
 }
 
+task_decrypt_password() {
+  local namespace=$1 cmd=$2 uuid=$3 ssh_key_pair_path=$4
+  [[ -n "${namespace}" ]] || { echo "[ERROR] 'namespace' is empty (${BASH_SOURCE[0]##*/}:${LINENO})" >&2; return 1; }
+  [[ -n "${cmd}"       ]] || { echo "[ERROR] 'cmd' is empty (${BASH_SOURCE[0]##*/}:${LINENO})" >&2; return 1; }
+  [[ -n "${uuid}"      ]] || { echo "[ERROR] 'uuid' is empty (${BASH_SOURCE[0]##*/}:${LINENO})" >&2; return 1; }
+  [[ -f "${ssh_key_pair_path}" ]] || { echo "[ERROR] 'ssh_key_file' is incorrect (${BASH_SOURCE[0]##*/}:${LINENO})" >&2; return 1; }
+
+  instance_password="$(task_show_password "$namespace" "$cmd" $uuid)" \
+      || { echo "[ERROR] encrypted password not available (${BASH_SOURCE[0]##*/}:${LINENO})" >&2; return 1; }
+  # ---
+  # :id: i-czulsyoj
+  # :encrypted_password: |
+  #  IAAMQn+4axLf/EO+dzQWhzDOBY/dkOCvDegiL8eS1VTs9MrwgA05vnCQ//EO
+  #  RaZPfWQLV4qhIjr1h4RUNQZ41Hs22lOztb2qpACuQzlfXpTKDp3YhZdOw/V8
+  #  hencQ02g8uziV5+Wpy1DLikkATMrRLurp2zsg7XQp/0mO9e16YzkzO9Xor3R
+  #  C+daGS/YeB8BZqQbwZgzWLnrgRm7q6zzKJoGxl67+RYHki/19n1gFQXNVyOi
+  #  /gZo1sdAPF/N2a18naOM4lpbjRr4R2u82mkgCMFYzRJew9yvOTGkFhwZamkX
+  #  9j+wU5Cvo0fDDpS4pXlRTvbmu9ESTmg0tSfFyuCgkA==
+
+  # assume :encrypted_password: is the last field and
```

Comment:
I changed the code in t.instance.base.sh to call this code, so it *will*
get autotested.  My purpose was to reduce duplicate code, but the
autotesting is a nice unintended side effect. :-)

Just noting, the two lines that introduce assumptions replaces this code in t.instance.base.sh:
```
  # 2: get encrypted_password value
  instance_password="$(echo "${instance_password}" | egrep '^  ' | sed 's,^  ,,')"
```
Which has its own assumptions.

Oh, I should have looked at your solution(!):
https://github.com/axsh/wakame-vdc/blob/20605e88d7496f5c532b72d8c0998b33e52eb377/dcmgr/script/decrypt_windows_password.sh#L35-L38

```
password=$(call_api \
  | awk -F '"encrypted_password":"' '{print $2}' \
  | awk -F '"' '{print $1}' \
  | sed 's/\\n/\n/g')
```

I can't tell what assumptions it has without jumping to the awk
documentation.

There must be many places in the code that need to parse out values
from yaml.  The best solution would be to have one trusted piece of
(bash) code that does this.

Options: (1) leave as is, since it will be autotested. (2) add a few more lines
of bash to remove the obvious assumptions.  (3) understand the awk solution and
maybe go with it. (4) ?

(The spec at http://www.yaml.org/spec/1.2/spec.html is depressingly long)
(this comment also got a little long :-)